solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,529 @@
+// SPDX-License-Identifier: SEE LICENSE IN LICENSE
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+import "@bancor/token-governance/contracts/ITokenGovernance.sol";
+
+import "../converter/interfaces/IConverterRegistry.sol";
+import "../converter/interfaces/IConverter.sol";
+
+import "../utility/ContractRegistryClient.sol";
+import "../utility/Owned.sol";
+import "../utility/Utils.sol";
+import "../utility/TokenHolder.sol";
+import "../utility/ReentrancyGuard.sol";
+
+import "../INetworkSettings.sol";
+import "../IBancorNetwork.sol";
+
+/**
+ * @dev This contract provides any user to trigger a network fee burning event
+ */
+contract VortexBurner is Owned, Utils, ReentrancyGuard, ContractRegistryClient {
+    using SafeMath for uint256;
+    using Math for uint256;
+    using SafeERC20 for IERC20;
+
+    struct Strategy {
+        address[][] paths;
+        uint256[] amounts;
+        address[] govPath;
+    }
+
+    struct NetNetworkTokenConversionAmounts {
+        uint256 amount;
+        uint256 incentiveFeeAmount;
+    }
+
+    // the mechanism is only designed to work with 50/50 standard pool converters
+    uint32 private constant STANDARD_POOL_RESERVE_WEIGHT = PPM_RESOLUTION / 2;
+
+    // the type of the standard pool converter
+    uint16 private constant STANDARD_POOL_CONVERTER_TYPE = 3;
```

Comment:
Maybe declare it `uint16 private immutable standardPoolConverterType`...
And then initialize it in the constructor using `StandardPoolConverterFactory.converterType()`.
?

Reply:
It's a cool idea, but it's currently not possible to call "static functions" by the contract name, but only via an instance

Reply:
Of course, but I didn't even mean to do it this way.
We can get the factory instance from the registry.

Reply:
Doesn't worth it